ztp/policygenerator/utils/filesHandler.go

```diff
@@ -55,24 +55,35 @@ func (fHandler *FilesHandler) ReadSourceFile(fileName string) ([]byte, error) {
 }
 
 func (fHandler *FilesHandler) ReadSourceCRFile(fileName string) ([]byte, error) {
-	var dir = ""
-	var err error = nil
-	var ret []byte
+	var (
+		gitDir   = ""
+		localDir = ""
+		err      error
+		fileByte []byte
+	)
 
-	ex, err := os.Executable()
+	// current working directory in git
+	gitDir, err = os.Getwd()
 	if err != nil {
 		return nil, err
 	}
-	dir = filepath.Dir(ex)
-	ret, err = fHandler.ReadFile(dir + "/" + SourceCRsPath + "/" + fileName)
+	sourceCRPathGit := gitDir + "/" + SourceCRsPath + "/" + fileName
 
-	// added fail safe for test runs as `os.Executable()` will fail for tests
+	// path of the local executable
+	localDir, err = os.Executable()
 	if err != nil {
-		dir, err = os.Getwd()
-		if err != nil {
-			return nil, err
+		return nil, err
+	}
+	dir := filepath.Dir(localDir)
+	sourceCRPathLocal := dir + "/" + SourceCRsPath + "/" + fileName
+
+	fileByte, err = os.ReadFile(sourceCRPathGit)
+	if errors.Is(err, os.ErrNotExist) {
+		fileByte, err := os.ReadFile(sourceCRPathLocal)
+		if errors.Is(err, os.ErrNotExist) {
+			return nil, errors.New(fileName + " is not found both in Git path: " + sourceCRPathGit + " and in the ztp container path: " + sourceCRPathLocal)
 		}
-		ret, err = fHandler.ReadFile(dir + "/" + SourceCRsPath + "/" + fileName)
+		return fileByte, nil
```

Comment:
By returning nil here we are masking any other potential error cases (eg file permissions) from the os.ReadFile. Any others are very unlikely but it seems like we should use:
```suggestion
		return fileByte, err
```